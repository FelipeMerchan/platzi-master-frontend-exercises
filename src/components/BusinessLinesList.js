import React from 'react'

import BusinessLine from './BusinessLine'
import './styles/BusinessLinesList.css'
import consultancyImage from '../images/consultoria.png'
import trainingImage from '../images/formacion.png'
import analysisImage from '../images/analisis.png'
import searchImage from '../images/investigacion.png'

function BusinessLinesList() {
  return (
    <section className="BusinessLinesList">
        <h1 className="BusinessLinesList__title">
          Soluciones corporativas <br />
          <span>lorem ipsum dolor sit amet, consectetuer adipiscing elit</span>
        </h1>
        <div>
          <BusinessLine
            style={{color: '#5c7e9d'}}
            image={consultancyImage}
            title={"Consultoría"}
            alt={"Consultoría"}
          />
          <BusinessLine
            style={{color: '#78b8b2'}}
            image={trainingImage}
            title={"Formación"}
            alt={"Formación"}
          />
          <BusinessLine
            style={{color: '#e5bb69'}}
            image={analysisImage}
            title={"Análisis"}
            alt={"Análisis"}
          />
          <BusinessLine
            style={{color: '#bb604f'}}
            image={searchImage}
            title={"Investigación"}
            alt={"Investigación"}
          />
        </div>
    </section>
  )
}

export default BusinessLinesList