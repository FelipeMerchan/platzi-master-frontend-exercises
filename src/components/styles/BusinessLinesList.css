.BusinessLinesList__title {
  margin: var(--space-3) 0;
  text-align: center;
  font-size: 68px;
  line-height: .7;
  color: #48504d;
  font-family: var(--font-3);
}

.BusinessLinesList__title span {
  font-size: 38px;
  color: #a6a7a2;
}

.BusinessLinesList > div {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-column-gap: var(--space-2);
    padding: var(--space-3) 0;
    border-top: 1px solid #dfdfdf;
}