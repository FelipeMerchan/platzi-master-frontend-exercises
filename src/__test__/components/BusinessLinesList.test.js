import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import BusinessLinesList from '../../components/BusinessLinesList'

describe('Probar el componente <BusinessLinesList />', () => {
  const businessLinesList = shallow(<BusinessLinesList />)
  test('Render del componente BusinessLinesList', () => {
    expect(businessLinesList.length).toEqual(1)
  })
  test('Render con el className BusinessLinesList__title', () => {
    expect(businessLinesList.find('h1').hasClass('BusinessLinesList__title')).toBe(true)
  })
  test('Validar el render de las 4 lÃ­neas de negocio', () => {
    expect(businessLinesList.find('div').children()).toHaveLength(4);
  });
})

describe('BusinessLinesList Snapshot', () => {
  const businessLinesList = create(<BusinessLinesList />)
  test('Comprobar la UI del componente BusinessLinesList', () => {
    expect(businessLinesList.toJSON()).toMatchSnapshot()
  })
})