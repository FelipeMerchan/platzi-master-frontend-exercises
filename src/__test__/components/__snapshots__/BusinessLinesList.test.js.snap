// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessLinesList Snapshot Comprobar la UI del componente BusinessLinesList 1`] = `
<section
  className="BusinessLinesList"
>
  <h1
    className="BusinessLinesList__title"
  >
    Soluciones corporativas 
    <br />
    <span>
      lorem ipsum dolor sit amet, consectetuer adipiscing elit
    </span>
  </h1>
  <div>
    <article
      className="BusinessLine"
    >
      <figure
        className="BusinessLine__imageContainer"
      >
        <img
          alt="Consultoría"
          className="BusinessLine__image"
          src="consultoria.png"
          width="200"
        />
      </figure>
      <div>
        <h3
          className="BusinessLine__name text-consultancy-color"
          style={
            Object {
              "color": "#5c7e9d",
            }
          }
        >
          Consultoría
        </h3>
        <p
          className="BusinessLine__description"
        >
          Lorem ipsum dolor sit amet, consectetuer aiscing elit.
        </p>
      </div>
    </article>
    <article
      className="BusinessLine"
    >
      <figure
        className="BusinessLine__imageContainer"
      >
        <img
          alt="Formación"
          className="BusinessLine__image"
          src="formacion.png"
          width="200"
        />
      </figure>
      <div>
        <h3
          className="BusinessLine__name text-consultancy-color"
          style={
            Object {
              "color": "#78b8b2",
            }
          }
        >
          Formación
        </h3>
        <p
          className="BusinessLine__description"
        >
          Lorem ipsum dolor sit amet, consectetuer aiscing elit.
        </p>
      </div>
    </article>
    <article
      className="BusinessLine"
    >
      <figure
        className="BusinessLine__imageContainer"
      >
        <img
          alt="Análisis"
          className="BusinessLine__image"
          src="analisis.png"
          width="200"
        />
      </figure>
      <div>
        <h3
          className="BusinessLine__name text-consultancy-color"
          style={
            Object {
              "color": "#e5bb69",
            }
          }
        >
          Análisis
        </h3>
        <p
          className="BusinessLine__description"
        >
          Lorem ipsum dolor sit amet, consectetuer aiscing elit.
        </p>
      </div>
    </article>
    <article
      className="BusinessLine"
    >
      <figure
        className="BusinessLine__imageContainer"
      >
        <img
          alt="Investigación"
          className="BusinessLine__image"
          src="investigacion.png"
          width="200"
        />
      </figure>
      <div>
        <h3
          className="BusinessLine__name text-consultancy-color"
          style={
            Object {
              "color": "#bb604f",
            }
          }
        >
          Investigación
        </h3>
        <p
          className="BusinessLine__description"
        >
          Lorem ipsum dolor sit amet, consectetuer aiscing elit.
        </p>
      </div>
    </article>
  </div>
</section>
`;
